Perhaps it doesn't work on windows?
Fresh install of this fork, with `MICRO_BATCH_SIZE = 64`, only uses a single GPU, even though `gpus = torch.cuda.device_count()` correctly detects 2 GPUs.
`ddp` remains `False` because WORLD_SIZE is not set.
Setting `PIPE_CHUNKS = 1` gives an error:
```
Traceback (most recent call last):
  File "E:\LLaMA-train\alpaca-lora\finetune.py", line 142, in <module>
    from torch.distributed.pipeline.sync import Pipe
  File "E:\LLaMA-train\installer_files\env\lib\site-packages\torch\distributed\pipeline\sync\__init__.py", line 9, in <module>
    from .pipe import Pipe, WithDevice
  File "E:\LLaMA-train\installer_files\env\lib\site-packages\torch\distributed\pipeline\sync\pipe.py", line 13, in <module>
    from torch.distributed.rpc import RRef
ImportError: cannot import name 'RRef' from 'torch.distributed.rpc' (E:\LLaMA-train\installer_files\env\lib\site-packages\torch\distributed\rpc\__init__.py)
```
Here are some relevant libs (transformers is also installed from the correct fork `git+https://github.com/kooshi/transformers.git@llama-parallelism`):
```
tokenizers              0.13.2
torch                   2.0.0
torchaudio              2.0.0
torchvision             0.15.0
tqdm                    4.65.0
transformers            4.28.0.dev0
```

--- a/finetune.py
+++ b/finetune.py
@@ -171,7 +171,7 @@ def train(
         val_data = None
 
     if not ddp and torch.cuda.device_count() > 1:
-        #keeps Trainer from trying it's own DataParallelism when more than 1 gpu is available
+        # keeps Trainer from trying its own DataParallelism when more than 1 gpu is available
         model.is_parallelizable = True
         model.model_parallel = True
         

--- a/finetune.py
+++ b/finetune.py
@@ -170,6 +170,11 @@ def train(
         train_data = data["train"].shuffle().map(generate_and_tokenize_prompt)
         val_data = None
 
+    if not ddp and torch.cuda.device_count() > 1:
+        #keeps Trainer from trying it's own DataParallelism when more than 1 gpu is available
+        model.is_parallelizable = True
+        model.model_parallel = True
+        
     trainer = transformers.Trainer(
         model=model,
         train_dataset=train_data,

--- a/finetune.py
+++ b/finetune.py
@@ -174,7 +174,7 @@ def train(
         # keeps Trainer from trying its own DataParallelism when more than 1 gpu is available
         model.is_parallelizable = True
         model.model_parallel = True
-        
+
     trainer = transformers.Trainer(
         model=model,
         train_dataset=train_data,

--- a/generate.py
+++ b/generate.py
@@ -28,9 +28,9 @@ def main(
     base_model: str = "",
     lora_weights: str = "tloen/alpaca-lora-7b",
 ):
-    assert base_model, (
-        "Please specify a --base_model, e.g. --base_model='decapoda-research/llama-7b-hf'"
-    )
+    assert (
+        base_model
+    ), "Please specify a --base_model, e.g. --base_model='decapoda-research/llama-7b-hf'"
 
     tokenizer = LlamaTokenizer.from_pretrained(base_model)
     if device == "cuda":

